### ISSUE TITLE
Just trying to setup a new react native project and web build fails

### ISSUE THREAD

**Issue by pmnzt** [expo/expo#27312]
### Summary

I first did 
```pnpm dlx create-expo-app AwesomeProject```
then ```cd AwesomeProject``` 
then
```
pnpm dlx expo install react-native-web react-dom @expo/metro-runtime```

then ```pnpm expo start```
then `w` to open the web.

But i got an empty page, the terminal log yields: 
```Web Bundling failed 34ms (node_modules/.pnpm/expo@50.0.8_@babel+core@7.23.9_@react-native+babel-preset@0.73.21/node_modules/expo/AppEntry.js)
Unable to resolve "../../App" from "node_modules/.pnpm/expo@50.0.8_@babel+core@7.23.9_@react-native+babel-preset@0.73.21/node_modules/expo/AppEntry.js"
```
other notes: 

During installing packages i got the following warn:
``` WARN  8 deprecated subdependencies found: @babel/plugin-proposal-async-generator-functions@7.20.7, @babel/plugin-proposal-class-properties@7.18.6, @babel/plugin-proposal-nullish-coalescing-operator@7.18.6, @babel/plugin-proposal-numeric-separator@7.18.6, @babel/plugin-proposal-object-rest-spread@7.20.7, @babel/plugin-proposal-optional-catch-binding@7.18.6, @babel/plugin-proposal-optional-chaining@7.21.0, @npmcli/move-file@1.1.2```

### SDK Version

0.17.6

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: Linux 6.1 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
      Shell: 5.2.15 - /bin/bash
    Binaries:
      Node: 18.17.1 - ~/Downloads/Nodejs/node-v18.17.1-linux-x64/bin/node
      Yarn: 1.22.19 - ~/Downloads/Nodejs/node-v18.17.1-linux-x64/bin/yarn
      npm: 9.6.7 - ~/Downloads/Nodejs/node-v18.17.1-linux-x64/bin/npm
    SDKs:
      Android SDK:
        API Levels: 31, 33
        Build Tools: 30.0.3, 33.0.0
    npmPackages:
      expo: ~50.0.8 => 50.0.8 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
      react-native-web: ~0.19.6 => 0.19.10 
    Expo Workflow: managed

### Minimal reproducible example

https://github.com/pmnzt/react-native-expo-reproduce

**Comment by leetdavid** [comment]
I'm having the same problem. I tried using expo years ago and it still feels like I can't even start a base project properly even now.

**Comment by pmnzt** [comment]
> I'm having the same problem. I tried using expo years ago and it still feels like I can't even start a base project properly even now.

yeah, that's unfortunate. me too have tried expo in the past but things did really work then, so i am trying now, here is the issue.
All we need is to get a simple demo app to work at least. or understand what's going on.

**Comment by FleetAdmiralJakob** [comment]
Same problem. Any fix?

**Comment by JavanShen** [comment]
Same problem after `pnpm start`

**Comment by FleetAdmiralJakob** [comment]
I was using the tabs example and in this template the `"@babel/runtime": "^7.24.0",` seems missing. Just install that and everything seems to be fine again.

**Comment by byCedric** [comment]
For now, I'll close this issue as it doesn't provide a specific case where isolated modules fails. If you have concrete examples of dependency imports that fail in isolated modules mode, feel free to open a new issue.
